corehq/apps/custom_data_fields/models.py

```diff
@@ -155,6 +155,26 @@ def validate_custom_fields(custom_fields):
 
         return validate_custom_fields
 
+    def get_model_and_uncategorized(self, data_dict):
+        """
+        Splits data_dict into two dictionaries:
+        one for data which matches the model and one for data that doesn't
+        Does not include reserved fields.
+        """
+        if not data_dict:
+            return {}, {}
+        model_data = {}
+        uncategorized_data = {}
+        slugs = [field.slug for field in self.sqlfield_set]
```

Comment:
Not sure how many fields there will be (probably doesn't matter if very few), but `k in <set>` is O(1) where for a list it's O(n).
```suggestion
        slugs = {field.slug for field in self.sqlfield_set}
```